### PR TITLE
Dev 1.1.2

### PR DESCRIPTION
**Pull request by visvel**

### CODE REVIEW

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 1]
```diff
@@ -710,6 +710,40 @@ function Start-CopyJob(
...
+    [string] $JobName,
+    [scriptblock] $ScriptBlock
+    )
+{
+	$Jobs       = @()
+	$Sessions   = @()
+	$SessionIds = @()
+
+	if ($ClusterNodes.Count -ge 1) {
+					$Sessions = New-PSSession -ComputerName $ClusterNodes
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Check your spaces v. tabs editor setting?

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 2]
```diff
@@ -710,6 +710,40 @@ function Start-CopyJob(
     }
 }
 
+#
+# Utility wrapper for invoking commands by opening sessions 
+# for each of the cluster nodes and preserving the session 
+# to be deleted after use.
+#
+
+function New-InvokeCommand (
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: We should keep using the Invoke verb, some different noun like SddcCommonCommand. "New" isn't right in this context, thats creation/instantiation, not alternate/different.

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 3]
```diff
@@ -710,6 +710,40 @@ function Start-CopyJob(
...
+    [string[]] $ClusterNodes = @(),
+    [string] $JobName,
+    [scriptblock] $ScriptBlock
+    )
+{
+	$Jobs       = @()
+	$Sessions   = @()
+	$SessionIds = @()
+
+	if ($ClusterNodes.Count -ge 1) {
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: what if -eq 1? I agree validation could be good, suggest doing that explicitly up front. I believe making ClusterNodes a mandatory parameter would enforce non-null (which is what you want).

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_functions_advanced_parameters?view=powershell-6

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 (hunk 1 above)
Comment (visvel) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Sure. Will fix it.

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 (hunk 2 above)
Comment (visvel) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Invoke-SddcCommonCommand sounds good?

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 (hunk 3 above)
Comment (dl2n) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Sorry, meant -eq 0 ...

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 [hunk 4]
```diff
@@ -710,6 +710,40 @@ function Start-CopyJob(
...
+					$Sessions = New-PSSession -ComputerName $ClusterNodes
+	}
+
+	$Jobs = Invoke-Command -Session $Sessions -AsJob -JobName $JobName -ScriptBlock $ScriptBlock
```
Comment (dl2n) [inline PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: How many jobs are returned?

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 (hunk 2 above)
Comment (dl2n) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Predict my answer, but I was intending that as a suggestion, not strictly a prescription ... sure, though.

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 (hunk 3 above)
Comment (visvel) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: Agreed.

File: PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1 (hunk 4 above)
Comment (visvel) [reply PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1]: one job, similar to what was returning from Invoke-Command output

**Pull request closed without merging**